README.md
# FooTable #

**Contributors:** bradvin

**Tags:** tables,responsive,jquery

**Requires at least:** 3.5

**Tested up to:** 3.6

**Stable tag:** 0.3

**License:** GPLv2 or later

**License URI:** http://www.gnu.org/licenses/gpl-2.0.html


**FooTable's goal is simple :** to make HTML tables look awesome on all devices!


## Description ##

Have you ever wanted to show a lot of data in a table, but hate how badly it scales on smaller mobile devices? FooTable solves this problem by hiding certain columns on smaller devices, but still allowing the user to expand each row to see the columns that were hidden

[FooTable WordPress Homepage](http://fooplugins.com/plugins/footable-lite/) |
[FooTable WordPress Documentation](http://fooplugins.com/footable-lite/documentation/) |
[FooTable jQuery Homepage](http://fooplugins.com/plugins/footable-jquery/) |
[FooTable jQuery GitHub Repo](https://github.com/bradvin/FooTable) |
[FooTable jQuery Demos & Docs](http://fooplugins.com/footable-demos/)

**features**

*	Make your existing HTML tables responsive
*	Target only the tables you want using a CSS selector
*	Add sorting, filtering and paging to your tables
*	Automatically hide certain number of columns on tablets and phones
*	Set custom breakpoint sizes
*   Choose from 3 table themes or use your own CSS
*	Ability to include custom CSS
*   Works with TablePress

## Installation ##

1.	Upload or unzip the plugin to your `/wp-content/plugins/` folder
2.	Activate the plugin through the 'Plugins' menu in WordPress
3.	Configure FooTable from the settings page

## Screenshots ##

###1. Settings - General###
![Settings - General](http://s.wordpress.org/extend/plugins/footable/screenshot-1.png)

###2. Settings - Demo###
![Settings - Demo](http://s.wordpress.org/extend/plugins/footable/screenshot-2.png)

###3. Frontend Result###
![Frontend Result](http://s.wordpress.org/extend/plugins/footable/screenshot-3.png)


## Changelog ##

### 0.3 ###
* Major update! Using FooBox V2 jQuery
* Reworked the settings

### 0.2.1 ###
* Updated JS to fix minor bug

### 0.2 ###
* Initial Relase. First version.

## Frequently Asked Questions ##

### How do I override the CSS that FooTable applies to my tables? ###
Goto the settings page and disable the FooTable CSS, or add your own custom styling using the "Custom CSS" setting

## Upgrade Notice ##

Please upgrade to the latest version

changelog.txt
FooTable Changelog
================

> Last updated 21 Feb 2013

2013.02.12 - version 0.2
------------------------
 * Initial release

includes/demo.php
<?php
/**
 * Demo on FooTables Settings Page
 *
 * @author 	Brad Vincent
 * @package 	footable/includes
 * @version     1.0
 */

?>
<div class="footale-demo-container">
	<table class="table footable-demo" data-auto-columns="false" data-page-size="5">
		<thead>
		<tr>
			<th>
				First Name
			</th>
			<th data-hide="phone">
				Last Name
			</th>
			<th data-hide="phone,tablet">
				Job Title
			</th>
			<th data-hide="phone,tablet">
				DOB
			</th>
			<th data-hide="phone">
				Status
			</th>
		</tr>
		</thead>
		<tbody>
		<tr>
			<td>Isidra</td>
			<td><a href="#">Boudreaux</a></td>
			<td>Traffic Court Referee</td>
			<td data-value="78025368997">22 Jun 1972</td>
			<td data-value="1"><span class="status-metro status-active" title="Active">Active</span></td>
		</tr>
		<tr>
			<td>Shona</td>
			<td>Woldt</td>
			<td><a href="#">Airline Transport Pilot</a></td>
			<td data-value="370961043292">3 Oct 1981</td>
			<td data-value="2"><span class="status-metro status-disabled" title="Disabled">Disabled</span></td>
		</tr>
		<tr>
			<td>Granville</td>
			<td>Leonardo</td>
			<td>Business Services Sales Representative</td>
			<td data-value="-22133780420">19 Apr 1969</td>
			<td data-value="3"><span class="status-metro status-suspended" title="Suspended">Suspended</span></td>
		</tr>
		<tr>
			<td>Easer</td>
			<td>Dragoo</td>
			<td>Drywall Stripper</td>
			<td data-value="250833505574">13 Dec 1977</td>
			<td data-value="1"><span class="status-metro status-active" title="Active">Active</span></td>
		</tr>
		<tr>
			<td>Maple</td>
			<td>Halladay</td>
			<td>Aviation Tactical Readiness Officer</td>
			<td data-value="694116650726">30 Dec 1991</td>
			<td data-value="3"><span class="status-metro status-suspended" title="Suspended">Suspended</span></td>
		</tr>
		<tr>
			<td>Maxine</td>
			<td><a href="#">Woldt</a></td>
			<td><a href="#">Business Services Sales Representative</a></td>
			<td data-value="561440464855">17 Oct 1987</td>
			<td data-value="2"><span class="status-metro status-disabled" title="Disabled">Disabled</span></td>
		</tr>
		<tr>
			<td>Lorraine</td>
			<td>Mcgaughy</td>
			<td>Hemodialysis Technician</td>
			<td data-value="437400551390">11 Nov 1983</td>
			<td data-value="2"><span class="status-metro status-disabled" title="Disabled">Disabled</span></td>
		</tr>
		<tr>
			<td>Lizzee</td>
			<td><a href="#">Goodlow</a></td>
			<td>Technical Services Librarian</td>
			<td data-value="-257733999319">1 Nov 1961</td>
			<td data-value="3"><span class="status-metro status-suspended" title="Suspended">Suspended</span></td>
		</tr>
		<tr>
			<td>Judi</td>
			<td>Badgett</td>
			<td>Electrical Lineworker</td>
			<td data-value="362134712000">23 Jun 1981</td>
			<td data-value="1"><span class="status-metro status-active" title="Active">Active</span></td>
		</tr>
		<tr>
			<td>Lauri</td>
			<td>Hyland</td>
			<td>Blackjack Supervisor</td>
			<td data-value="500874333932">15 Nov 1985</td>
			<td data-value="3"><span class="status-metro status-suspended" title="Suspended">Suspended</span></td>
		</tr>
		</tbody>
	</table>
</div>

views/settings.php
<?php
/*
Default settings page used by Foo_PluginBase
*/
global $wp_version, $settings_data, $wp_settings_sections, $wp_settings_fields;

$summary = $settings_data['settings_summary'];
$tabs = $settings_data['settings_tabs'];
$plugin_slug = $settings_data['plugin_info']['slug'];

?>
<div class="wrap" id="<?php echo $plugin_slug; ?>-settings">
	<div id="icon-options-general" class="icon32">
            <br />
	</div>

	<h2><?php echo esc_html( get_admin_page_title() ); ?></h2>
	<?php
        echo $summary;
        //only show the settings messages if less than WP3.5
        if (version_compare($wp_version, '3.5') < 0) {
            settings_errors();
        }

        if ( !isset($wp_settings_sections) || !isset($wp_settings_sections[$plugin_slug]) )
            return;
    ?>
	<form action="options.php" method="post">
		<?php settings_fields($plugin_slug); ?>
                <?php
                if (!empty($tabs)) {
                    //we have tabs - woot!
                ?>
                <div style="float:left;height:16px;width:16px;"><!-- spacer for tabs --></div>
                <h3 class="nav-tab-wrapper">
                <?php
                    //loop through the tabs to render the actual tabs at the top
                    $first = true;
                    foreach ($tabs as $tab) {
                        $class = $first ? "nav-tab nav-tab-active" : "nav-tab";
                        echo "<a href='#{$tab['id']}' class='$class'>{$tab['title']}</a>";
                        if ($first) { $first = false; }
                    }
                ?>
                </h3>
                <?php
                    //now loop through the tabs to render the content containers
                    $first = true;
                    foreach ($tabs as $tab) {
                        $style = $first ? "" : "style='display:none'";

                        echo "<div class='nav-container' id='{$tab['id']}_tab' $style>";

						foreach ( (array) $wp_settings_sections[$plugin_slug] as $section ) {
							if (in_array($section['id'], $tab['sections'])) {
								echo "<h3>{$section['title']}</h3>\n";
								call_user_func($section['callback'], $section);
								if ( !isset($wp_settings_fields) || !isset($wp_settings_fields[$plugin_slug]) || !isset($wp_settings_fields[$plugin_slug][$section['id']]) )
									continue;
								echo '<table class="form-table">';
								do_settings_fields($plugin_slug, $section['id']);
								echo '</table>';
							}
						}

                        echo "</div>";
                        if ($first) { $first = false; }
                    }
                ?>
                <?php
                } else {
                    //no tabs so just render the sections
                    do_settings_sections($plugin_slug);
                }
                ?>
		<p class="submit">
			<input name="Submit" class="button-primary" type="submit" value="<?php _e('Save Changes', $plugin_slug); ?>" />
                        <input name="<?php echo $plugin_slug; ?>[reset-defaults]" onclick="return confirm('<?php _e('Are you sure you want to restore all settings back to their default values?', $plugin_slug); ?>');" class="button-secondary" type="submit" value="<?php _e('Restore Defaults', $plugin_slug); ?>" />
		</p>
	</form>
</div>

js/admin-settings.js
jQuery(document).ready(function($) {
    $.admin_tabs = {

        init : function() {
          $("a.nav-tab").click( function(e) {
              e.preventDefault();

              $this = $(this);

              $this.parents(".nav-tab-wrapper:first").find(".nav-tab-active").removeClass("nav-tab-active");
              $this.addClass("nav-tab-active");

              $(".nav-container:visible").hide();

              var hash = $this.attr("href");

              $(hash+'_tab').show();

	      if (hash == '#demo') {
		      if ($('.footable-demo').data('footable')) $('.footable-demo').data('footable').resize();
	      }

              //fix the referer so if changes are saved, we come back to the same tab
              var referer = $("input[name=_wp_http_referer]").val();
              if (referer.indexOf("#") >= 0) {
                referer = referer.substr(0, referer.indexOf("#"));
              }
              referer += hash;

              window.location.hash = hash;

              $("input[name=_wp_http_referer]").val(referer);
          });

          if (window.location.hash) {
            $('a.nav-tab[href="' + window.location.hash + '"]').click();
          }

          return false;
        }

    }; //End of admin_tabs

    $.admin_tabs.init();

    $('a.trigger-nav-tab').click(function(e) {
        var href = $(this).attr('href');
	    $('a.nav-tab[href="'+href+'"]').click();
    });

	$('.footable-device-selector > a').click(function(e) {
		e.preventDefault();
		var $this = $(this);
		$this.addClass('chosen').siblings('a').removeClass('chosen');
		var width = $this.data('width');
		$('.footale-demo-container').animate({ 'width': width }, function() {
            if ($('.footable-demo').data('footable')) $('.footable-demo').data('footable').resize();
		});
	});

});

includes/admin_settings.php
<?php

if (!class_exists('FooTable_Settings')) {

    class FooTable_Settings {

        /**
         *
         * @param $footable FooTable
         * @param $settings Foo_Plugin_Settings_v1_0
         */
        static function create_settings($footable, $settings) {

			$settings->add_tab( 'general', __('General', 'footable') );

			$help_html = '</h3><div class="foo-alert foo-alert-info">';
			$help_html .= '<h3>' . __( 'What is FooTable?', 'footable' ) . '</h3>';
			$help_html .= '<p>' . __( 'FooTable\'s goal is simple : to make HTML tables look awesome on all devices!', 'footable') . '</p>';
			$help_html .= '<p>' . __( 'Have you ever wanted to show a lot of data in a table, but hate how badly it scales on smaller mobile devices? FooTable solves this problem by hiding certain columns on smaller devices, but still allowing the user to expand each row to see the columns that were hidden.', 'footable') . '</p>';
			$help_html .= '<p><a class="trigger-nav-tab" href="#demo">' . __('Check out the demo to see it in action', 'footable') . '</a></p>';
			$help_html .= '</div>';

			$settings->add_section_to_tab( 'general', 'help', $help_html );

			$links_html = '</h3><div class="foo-alert foo-alert-warning footable-links">';
			$links_html .= '<h3>' . __( 'Useful FooTable Links', 'footable' ) . '</h3>';
			$links_html .= '<a target="_blank" href="' . FooTable::URL_DOCS . '"><strong>' . __('Getting Started Guide', 'footable') . '</strong></a>';
			$links_html .= '<a target="_blank" href="' . FooTable::URL_HOMEPAGE . '">' . __('Plugin Homepage', 'footable') . '</a>';
			$links_html .= '<a target="_blank" href="' . FooTable::URL_JQUERY . '">' . __('jQuery Plugin', 'footable') . '</a>';
			$links_html .= '<a target="_blank" href="' . FooTable::URL_GITHUB . '">' . __('jQuery Plugin on GitHub', 'footable') . '</a>';
			$links_html .= '</div>';

			$settings->add_section_to_tab( 'general', 'links', $links_html );

			$settings->add_section_to_tab( 'general', 'tables', __('FooTable Setup', 'footable') );

			$settings->add_setting( array(
				'id'      => 'selector',
				'title'   => __( 'Attach to tables', 'footable' ),
				'desc'    => __( 'Attach FooTable to tables with a CSS selector. Use this to target only certain HTML tables in your site.<br />The most common scenario is using a class name e.g. <code>.footable</code>. You can also target all tables in your site by using <code>table</code>', 'footable' ),
				'default' => '.footable',
				'type'    => 'text',
				'section' => 'tables',
				'tab'     => 'general'
			) );

			if (class_exists('TablePress')) {
				$settings->add_setting( array(
					'id'      => 'tablepress',
					'title'   => __( 'Attach to TablePress tables', 'footable' ),
					'desc'    => __( 'Automatically attach FooTable to all tables you create using TablePress.', 'footable' ),
					'default' => 'on',
					'type'    => 'checkbox',
					'section' => 'tables',
					'tab'     => 'general'
				) );
			}

			$settings->add_setting( array(
				'id'      => 'enable_sorting',
				'title'   => __( 'Enable Sorting', 'footable' ),
				'desc'    => __( 'Enable column sorting on all FooTables.<br />To disable sorting on a specific table, add <code>data-sort="false"</code> to your table element.', 'footable' ),
				'default' => 'on',
				'type'    => 'checkbox',
				'section' => 'tables',
				'tab'     => 'general'
			) );

			$settings->add_setting( array(
				'id'      => 'enable_filtering',
				'title'   => __( 'Enable Filtering', 'footable' ),
				'desc'    => __( 'Enable column filtering on all FooTables.<br />To disable sorting on a specific table, add <code>data-filter="false"</code> to your table element.', 'footable' ),
				'default' => 'on',
				'type'    => 'checkbox',
				'section' => 'tables',
				'tab'     => 'general'
			) );

			$settings->add_setting( array(
				'id'      => 'enable_pagination',
				'title'   => __( 'Enable Pagination', 'footable' ),
				'desc'    => __( 'Enable pagination on all FooTables.<br />To disable sorting on a specific table, add <code>data-page="false"</code> to your table element.', 'footable' ),
				'default' => 'on',
				'type'    => 'checkbox',
				'section' => 'tables',
				'tab'     => 'general'
			) );

//			$settings->add_setting( array(
//				'id'      => 'check_parent_width',
//				'title'   => __( 'Enable Parent Checking', 'footable' ),
//				'desc'    => __( 'When the FooTable is resized, the parent element\'s width is checked to make sure the FooTable breakpoints are triggered correctly.', 'footable' ),
//				'default' => 'on',
//				'type'    => 'checkbox',
//				'section' => 'tables',
//				'tab'     => 'general'
//			) );

			$settings->add_tab( 'breakpoints', __('Breakpoints', 'footable') );

			$breakpoint_html = '</h3><div class="foo-alert foo-alert-info">';
			$breakpoint_html .= '<h3>' . __( 'Breakpoint Help', 'footable' ) . '</h3>';
			$breakpoint_html .= '<p>' . __( 'Breakpoints are the heart and soul of FooTable. Whenever your site is viewed on a mobile device, or if the browser window is resized, FooTable checks the width of the table. If that width is smaller than the width of a breakpoint, certain columns in the table will be hidden.', 'footable') . '</p>';
			$breakpoint_html .= '<p>' . __( 'FooTable has two default breakpoints : <strong>tablet</strong> and <strong>phone</strong>. You can change the default size of these breakpoints below, so that they match your site\'s theme.', 'footable') . '</p>';
			$breakpoint_html .= '<p><a target="_blank" href="' . FooTable::URL_DOCS . '">' . __('Read more at the plugin documentation', 'footable') . '</a></p>';
			$breakpoint_html .= '</div>';

			$settings->add_section_to_tab( 'breakpoints', 'breakpoints', $breakpoint_html );

			$settings->add_setting( array(
				'id'      => 'breakpoint_tablet',
				'title'   => __( 'Tablet Breakpoint', 'footable' ),
				'desc'    => __( 'The width of the tablet breakpoint', 'footable' ),
				'default' => '768',
				'type'    => 'text',
				'section' => 'breakpoints',
				'tab'     => 'breakpoints',
				'class'   => 'short_input'
			) );

			$settings->add_setting( array(
				'id'      => 'breakpoint_phone',
				'title'   => __( 'Phone Breakpoint', 'footable' ),
				'desc'    => __( 'The width of the phone breakpoint', 'footable' ),
				'default' => '320',
				'type'    => 'text',
				'section' => 'breakpoints',
				'tab'     => 'breakpoints',
				'class'   => 'short_input'
			) );

			$settings->add_section_to_tab( 'breakpoints', 'columns', __('Column Visibility', 'footable') );

			$settings->add_setting( array(
				'id'      => 'columns_tablet',
				'title'   => __( '# Columns visible on tablet', 'footable' ),
				'desc'    => __( 'Max number of columns that are visible when the <strong>tablet</strong> breakpoint is triggered', 'footable' ),
				'default' => '4',
				'type'    => 'text',
				'section' => 'columns',
				'tab'     => 'breakpoints',
				'class'   => 'short_input'
			) );

			$settings->add_setting( array(
				'id'      => 'columns_phone',
				'title'   => __( '# Columns visible on phone', 'footable' ),
				'desc'    => __( 'Max number of columns that are visible when the <strong>phone</strong> breakpoint is triggered', 'footable' ),
				'default' => '2',
				'type'    => 'text',
				'section' => 'columns',
				'tab'     => 'breakpoints',
				'class'   => 'short_input'
			) );

			$settings->add_setting( array(
				'id'      => 'manual_columns',
				'title'   => __( 'Manual Column Visibility', 'footable' ),
				'desc'    => __( 'You can manually define which table columns will be visible by adding data attributes to your table HTML.', 'footable' ) . ' <a target="_blank" href="' . FooTable::URL_DOCS . '">' .  __('Read the documentation for more details', 'footable') . '</a>.',
				'type'    => 'checkbox',
				'section' => 'columns',
				'tab'     => 'breakpoints'
			) );

//			$settings->add_setting( array(
//				'id'      => 'columns_ignore',
//				'title'   => __( 'Always show columns', 'footable' ),
//				'desc'    => __( 'The names of the columns you always want to show on all devices. Please note that this will override the max columns set above.', 'footable' ),
//				'default' => '',
//				'type'    => 'text',
//				'section' => 'breakpoints',
//				'tab'     => 'general'
//			) );

			$settings->add_tab( 'looknfeel', __('Look &amp; Feel', 'footable') );

			$theme_html = '</h3><div class="foo-alert foo-alert-info">';
			$theme_html .= '<h3>' . __( 'FooTable Theme', 'footable' ) . '</h3>';
			$theme_html .= '<p>' . __( 'FooTable is built to look great with BootStrap out of the box, and the default theme adds no table styling and rather uses the BootStrap table styles.', 'footable') . '</p>';
			$theme_html .= '<p>' . __( 'However, we do have two table themes you can use if your theme does not use BootStrap.', 'footable') . '</p>';
			$theme_html .= '<p><a target="_blank" href="' . FooTable::URL_DOCS . '">' . __('Read more at the plugin documentation', 'footable') . '</a></p>';
			$theme_html .= '</div>';

			$settings->add_section_to_tab( 'looknfeel', 'theme', $theme_html );

			$theme_choices = array(
				'bootstrap' => __('Default', 'footable'),
				'metro' => __('Metro', 'footable'),
				'original' => __('Original FooTable Theme', 'footable'),
				'none' => __('None (use custom CSS)', 'footable'),
			);

			$settings->add_setting( array(
				'id'      => 'theme',
				'title'   => __( 'FooTable Theme', 'footable' ),
				'desc'	  => __( 'If you choose to use the default theme, then we assume you already have the necessary BootStrap CSS included in your theme.', 'footable'),
				'default' => 'bootstrap',
				'type'    => 'radio',
				'choices' => $theme_choices,
				'tab'     => 'looknfeel',
				'section' => 'theme'
			) );

			$settings->add_setting( array(
				'id'      => 'custom_css',
				'title'   => __( 'Custom CSS', 'footable' ),
				'desc'    => __( 'Add your own custom CSS styles', 'footable' ),
				'default' => '',
				'type'    => 'textarea',
				'tab'     => 'looknfeel',
				'section' => 'theme',
				'class'   => 'medium_textarea'
			) );

			$settings->add_section_to_tab( 'advanced', 'advanced', __('Advanced', 'footable') );

			$settings->add_tab( 'advanced', __('Advanced', 'footable') );

			$settings->add_section_to_tab( 'advanced', 'js', __('Javascript', 'footable') );

			$settings->add_setting( array(
				'id'      => 'custom_js_before',
				'title'   => __( 'Custom Javascript (Before)', 'footable' ),
				'desc'    => __( 'Call any custom JS before FooTable is initialized. (Only to be used by developers!)', 'footable' ),
				'default' => '',
				'type'    => 'textarea',
				'tab'     => 'advanced',
				'section' => 'js',
				'class'   => 'medium_textarea'
			) );

			$settings->add_setting( array(
				'id'      => 'custom_js_after',
				'title'   => __( 'Custom Javascript (After)', 'footable' ),
				'desc'    => __( 'Alter the way FooTable works by hooking into the built-in events, using custom javascript code. (Only to be used by developers!)', 'footable' ),
				'default' => '',
				'type'    => 'textarea',
				'tab'     => 'advanced',
				'section' => 'js',
				'class'   => 'medium_textarea'
			) );

			$settings->add_setting( array(
				'id'      => 'scripts_in_footer',
				'title'   => __( 'Scripts In Footer', 'footable' ),
				'desc'    => __( 'Load the javascript files in the site footer (for better performance). This requires the theme to have the wp_footer() hook in the appropriate place', 'footable' ),
				'type'    => 'checkbox',
				'section' => 'js',
				'tab'     => 'advanced'
			) );

			$settings->add_section_to_tab( 'advanced', 'debug', __('Debug', 'footable') );

			$settings->add_setting( array(
				'id'      => 'enable_debug',
				'title'   => __( 'Enable Debug Mode', 'footable' ),
				'desc'    => __( 'When debug is enabled, FooTable will write to the console log so you can debug any problems.<br />We also show an extra debug information tab on this settings page', 'footable' ),
				'type'    => 'checkbox',
				'section' => 'debug',
				'tab'     => 'advanced'
			) );

			$settings->add_tab( 'demo', __('Demo', 'footable') );

			$device_selector = '<div class="footable-device-selector">' .
'	<a title="Phone (320px)" class="phone" data-width="300px" href="#mobile"></a>' .
'	<a title="Tablet (768px)" class="tablet" data-width="750px" href="#tablet"></a>' .
'	<a title="Desktop (1024px)" class="desktop" data-width="1024px" href="#desktop"></a>' .
'</div>';

			$settings->add_setting( array(
				'id'      => 'demo_js',
				'title'   => $device_selector,
				'type'    => 'demo',
				'tab'     => 'demo'
			) );

			if ( $footable->options()->is_checked( 'enable_debug' ) ) {
				$settings->add_tab( 'debug', __('Debug Info', 'footable') );

				$settings->add_setting( array(
					'id'      => 'debug_output',
					'title'   => __( 'Debug Information', 'footable' ),
					'default' => 'off',
					'type'    => 'debug_output',
					'tab'     => 'debug'
				) );
			}
        }
    }
}

js/footable-wordpress.debug.js
jQuery.fn.footableAttr = function (tabletCount, phoneCount) {
    return this.each(function () {
        var $t = $(this);
        if ($t.data("auto-columns") !== false) {
            $t.find("thead th:gt(" + tabletCount + ")").attrAppendWithComma("data-hide", "tablet");
            $t.find("thead th:gt(" + phoneCount + ")").attrAppendWithComma("data-hide", "phone");
        }
    })
};

jQuery.fn.footableFilter = function (searchText) {
    return this.each(function () {
        var $t = $(this);
        if (!$t.data("filter") && $t.data("filter") !== false) {
            $t.data("filter-text-only","true")
                .before("<div class=\"footable-filter-container\"><input placeholder=\""+searchText+"\" style=\"float:right\" type=\"text\" class=\"footable-filter\" /></div>");
        }
    })
};

jQuery.fn.footablePager = function () {
    return this.each(function () {
        var $t = $(this);
        if ($t.data("page") !== false) {
            var $pager = $("<tfoot class=\"hide-if-no-paging\"><tr><td><div class=\"pagination pagination-centered\"></div></td></tr></tfoot>");
            $pager.find("td").attr("colspan", $t.find("thead th").length);
            $t.find("tbody:last").after($pager);
        }
    })
};

class.footable.php
<?php
/**
 * FooTable
 *
 * @package   FooTable
 * @author    Brad Vincent <brad@fooplugins.com>
 * @license   GPL-2.0+
 * @link      http://fooplugins.com/plugins/footable-lite/
 * @copyright 2013 FooPlugins LLC
 */


if (!class_exists('FooTable')) {

	define('FOOTABLE_PATH', plugin_dir_path( __FILE__ ));

	// Includes
	require_once( FOOTABLE_PATH . 'includes/Foo_Plugin_Base.php' );
	require_once( FOOTABLE_PATH . 'includes/admin_settings.php' );

	class FooTable extends Foo_Plugin_Base_v1_1 {

		const JS = 'footable.min.js';
		const JS_SORT = 'footable.sort.min.js';
		const JS_FILTER = 'footable.filter.min.js';
		const JS_PAGINATE = 'footable.paginate.min.js';

		const CSS = 'footable.core.min.css';
		const CSS_BOOTSTRAP = 'bootstrap.2.3.1.css';
		const CSS_METRO = 'footable.metro.min.css';
		const CSS_STANDALONE = 'footable.standalone.min.css';

		const URL_HOMEPAGE = 'http://fooplugins.com/plugins/footable-lite/';
		const URL_GITHUB = 'https://github.com/bradvin/FooTable';
		const URL_JQUERY = 'http://fooplugins.com/plugins/footable-jquery/';
		const URL_JQUERY_DEMOS = 'http://fooplugins.com/footable-demos/';
		const URL_DOCS = 'http://fooplugins.com/footable-lite/documentation/';

		function __construct($file) {
			$this->init(  $file, 'footable', '0.3', 'FooTable' );
			add_action( 'init', array($this, 'init_footable') );
		}

		function init_footable() {
			//TODO: translations
			//add_action('plugins_loaded', array($this, 'load_text_domain') );

			if (is_admin()) {
				add_action( $this->plugin_slug . '-admin_create_settings', array('FooTable_Settings', 'create_settings'), 10, 2 );
				add_action( $this->plugin_slug . '-settings_custom_type_render', array($this, 'custom_admin_settings_render') );
				add_action( 'admin_head', array($this, 'admin_inline_content') );
				add_filter( $this->plugin_slug . '-settings_summary', array($this, 'admin_settings_summary') );
				add_action( $this->plugin_slug . '-admin_print_styles', array($this, 'add_styles') );
				add_action( $this->plugin_slug . '-admin_print_scripts', array($this, 'add_scripts') );
			} else {
				$this->frontend_init();
			}
		}

		function admin_settings_summary() {
			$html = __('For more info about FooTable, please visit the <a href="%s" target="_blank">FooTable homepage</a>.', 'footable');
			return sprintf($html, self::HOMEPAGE_URL);
		}

		function custom_admin_settings_render( $args = array() ) {
			$type = '';

			extract( $args );

			if ($type == 'debug_output') {
				echo '</td></tr><tr valign="top"><td colspan="2">';
				$this->render_debug_info();
			} else if ( $type == 'demo') {
				echo '</td></tr><tr valign="top"><td colspan="2">';
				$this->render_demo();
			}
		}

		function generate_javascript($debug = false) {
			$js = '/* FooTable init code */
';
			$no_js = true;

			$js .= '
var $FOOTABLE = $FOOTABLE || {};
(function( $FOOTABLE, $, undefined ) {

	jQuery.fn.attrAppendWithComma=function(a,b){var c;return this.each(function(){c=$(this),void 0!==c.attr(a)&&""!=c.attr(a)?c.attr(a,c.attr(a)+","+b):c.attr(a,b)})};jQuery.fn.footableAttr=function(a,b){return this.each(function(){var c=$(this);c.data("auto-columns")!==!1&&(c.find("thead th:gt("+a+")").attrAppendWithComma("data-hide","tablet"),c.find("thead th:gt("+b+")").attrAppendWithComma("data-hide","phone"))})},jQuery.fn.footableFilter=function(a){return this.each(function(){var b=$(this);b.data("filter")||b.data("filter")===!1||b.data("filter-text-only","true").before(\'<div class="footable-filter-container"><input placeholder="\'+a+\'" style="float:right" type="text" class="footable-filter" /></div>\')})},jQuery.fn.footablePager=function(){return this.each(function(){var a=$(this);if(a.data("page")!==!1){var b=$(\'<tfoot class="hide-if-no-paging"><tr><td><div class="pagination pagination-centered"></div></td></tr></tfoot>\');b.find("td").attr("colspan",a.find("thead th").length),a.find("tbody:last").after(b)}})};

	$FOOTABLE.init = function() {
		var $tables;
';
			$breakpoint_tablet = $this->options()->get_int( 'breakpoint_tablet', 768 );
			$breakpoint_phone = $this->options()->get_int( 'breakpoint_phone', 320 );

			$columns_tablet = $this->options()->get_int( 'columns_tablet', 4 ) - 1;
			$columns_phone = $this->options()->get_int( 'columns_phone', 2 ) - 1;
			$manual_columns = $this->options()->is_checked( 'manual_columns' );
			$tablepress = $this->options()->is_checked('tablepress', true);

			//get custom JS (Before) from the settings page
			$custom_js_before = $this->options()->get( 'custom_js_before' );

			if ( !empty($custom_js_before) ) {
				$no_js = false;
				$js .= '    ' . $custom_js_before . '
';
			}

			$selector = $this->options()->get( 'selector', '.footable' );
			if ($tablepress) {
				$selector .= ', .tablepress';
			}
			if ( $this->screen()->is_plugin_settings_page() ) {
				$selector .= ', .footable-demo';
			}

			if ( !empty( $selector ) ) {
				$no_js = false;
				$js .= '		$tables = $("'.$selector.'");
';

				if ( !$manual_columns ) {
					$js .= '
		$tables.footableAttr('.$columns_tablet.','.$columns_phone.');
';
				}

				$js .= '		$tables.footableFilter("' . __('search','footable') . '")
			.footablePager()
			.footable( { breakpoints: { phone: '.$breakpoint_phone.', tablet: '.$breakpoint_tablet.' } });
';
			}

			//get custom JS from the settings page
			$custom_js_after = $this->options()->get( 'custom_js_after' );

			if ( !empty($custom_js_after) ) {
				$no_js = false;
				$js .= '    ' . $custom_js_after . '
';
			}

			$js .= '
	};
}( $FOOTABLE, jQuery ));

jQuery(function($) {
	$FOOTABLE.init();
});
';

			if ($no_js) { return ''; }

			return $js;
		}

		function render_debug_info() {
			echo '<strong>Javascript:<br /><pre>';
			echo htmlentities($this->generate_javascript(true));
			echo '</pre><br />Settings:<br /><pre>';
			print_r( get_option( $this->plugin_slug ) );
			echo '</pre>';
		}

		function render_demo() {
			require_once "includes/demo.php";
		}

		function frontend_init() {
			add_action( 'wp_enqueue_scripts', array($this, 'add_styles'), 12 );
			add_action( 'wp_enqueue_scripts', array($this, 'add_scripts'), 12 );

			$where = 'wp_head';

			if ( $this->options()->is_checked( 'scripts_in_footer' ) ) { $where = 'wp_print_footer_scripts'; }

			add_action($where, array($this, 'inline_dynamic_js') );

			add_action('wp_print_styles', array($this, 'inline_dynamic_css') );
		}

		function add_styles() {
			if (is_admin() && $this->screen()->is_plugin_settings_page()) {
				$this->register_and_enqueue_css(self::CSS_BOOTSTRAP);
			}

			//enqueue footable CSS
			$this->register_and_enqueue_css(self::CSS);

			$theme = $this->options()->get('theme', 'bootstrap');
			if ($theme === 'metro') {
				$this->register_and_enqueue_css(self::CSS_METRO);
			} else if ($theme === 'original') {
				$this->register_and_enqueue_css(self::CSS_STANDALONE);
			}
		}

		function add_scripts() {
			//put JS in footer?
			$infooter = !is_admin() && $this->options()->is_checked( 'scripts_in_footer' );

			//enqueue core JS
			$this->register_and_enqueue_js(self::JS, array('jquery'), false, $infooter);

			//enqueue sorting
			if ($this->options()->is_checked('enable_sorting', true)) {
				$this->register_and_enqueue_js(self::JS_SORT, array('jquery'), false, $infooter);
			}

			//enqueue filtering
			if ($this->options()->is_checked('enable_filtering', true)) {
				$this->register_and_enqueue_js(self::JS_FILTER, array('jquery'), false, $infooter);
			}

			//enqueue paging
			if ($this->options()->is_checked('enable_pagination', true)) {
				$this->register_and_enqueue_js(self::JS_PAGINATE, array('jquery'), false, $infooter);
			}
		}

		function admin_inline_content() {
			if ($this->screen()->is_plugin_settings_page()) {
				$this->inline_dynamic_css();
				$this->inline_dynamic_js();
			}
		}

		function inline_dynamic_js() {
			$footable_js = $this->generate_javascript();

			echo '<script type="text/javascript">' . $footable_js . '</script>';
		}

		function inline_dynamic_css() {

			//get custom CSS from the settings page
			$custom_css = $this->options()->get( 'custom_css', '' );

			if (class_exists('TablePress')) {
				$custom_css .= '.tablepress thead th div { float:left; }';
			}

			if (empty($custom_css)) return;

			echo '<style type="text/css">
	' . $custom_css;
			echo '
</style>';
		}
	}
}